File: PRAK305-2310817220016-NailaHanifah.py
def konversi_detik(total_detik):
    if total_detik < 0:
        return "Input tidak valid."
    elif total_detik < 86400:
        jam = total_detik // 3600
        sisa_detik = total_detik % 3600
        menit = sisa_detik // 60
        detik = sisa_detik % 60
        return "{:02d}:{:02d}:{:02d}".format(jam, menit, detik)
    else:
        hari = total_detik // 86400
        sisa_detik_hari = total_detik % 86400
        jam = sisa_detik_hari // 3600
        sisa_detik = sisa_detik_hari % 3600
        menit = sisa_detik // 60
        detik = sisa_detik % 60
        return "{} hari {:02d}:{:02d}:{:02d}".format(hari, jam, menit, detik)

total_detik = int(input("Masukkan jumlah detik: "))
print(konversi_detik(total_detik))

File: PRAK305-2310817220016-NailaHanifah.c
#include <stdio.h>

int main() {
    int totalDetik, sisaDetik, jam, menit, detik;

    printf("Masukkan jumlah detik: ");
    scanf("%d", &totalDetik);

    jam = totalDetik / 3600;
    sisaDetik = totalDetik % 3600;
    menit = sisaDetik / 60;
    detik = sisaDetik % 60;

    if (totalDetik < 0) {
        printf("Masukan tidak valid.\n");
    } else if (totalDetik < 86400) {
        printf("Output: %02d:%02d:%02d\n", jam, menit, detik);
    } else {
        int hari = totalDetik / 86400;
        int sisaDetikHari = totalDetik % 86400;
        jam = sisaDetikHari / 3600;
        sisaDetik = sisaDetikHari % 3600;
        menit = sisaDetik / 60;
        detik = sisaDetik % 60;
        printf("Output: %d hari %02d:%02d:%02d\n", hari, jam, menit, detik);
    }

    return 0;
}
